### PR TITLE
[MOB-4323] Restore default browser search promo

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-4323]

## Context

`DefaultBrowserViewController` and its `minSearchCountToTrigger` threshold (50 searches) were defined but never wired to an actual presentation path — the promo was never shown.

#1102 restored the search-count notification infrastructure (`searchesCounterChanged`) that this work depends on.

## Approach

- Observe `.searchesCounterChanged` in `BrowserViewController` and re-evaluate eligibility on every count change, as well as on `browserDidBecomeActive` to catch users already above the threshold
- Reuse the existing `DefaultBrowserDelegate.defaultBrowserDidShow` callback (already sets `PrefsKeys.IntroSeen`) as the "mark as shown" mechanism — no new prefs key introduced
- Made `searchesCounterChanged` `public` so the `Client` module can observe it (was `internal`, consistent with the neighbouring `searchSettingsChanged` which is already `public`)
- Extracted eligibility logic into a static pure function `isEligibleForEcosiaDefaultBrowserSearchPromo` to keep it unit-testable without spinning up a real `BrowserViewController`
- Added unit tests covering threshold boundary conditions, already-default-browser guard, and already-shown guard

## Other

The existing `ToggleDefaultBrowserPromo` debug setting already resets `PrefsKeys.IntroSeen`, so no changes to `EcosiaDebugSettings` were needed beyond confirming the reset path works end-to-end.

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator for both iPhone and iPad
- [x] I wrote Unit Tests that confirm the expected behaviour
- [x] I added the `// Ecosia:` helper comments where needed
- [ ] I made sure that any change to the Analytics events included in PR won't alter current analytics (e.g. new users, upgrading users)
- [ ] I included documentation updates to the coding standards or Confluence doc, when needed

[MOB-4323]: https://ecosia.atlassian.net/browse/MOB-4323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: firefox-ios/Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift [hunk 1]
```diff
@@ -21,8 +21,54 @@ extension BrowserViewController: HomepageViewControllerDelegate {
...
+        // Promo completion is tracked when scheduling presentation (`ecosiaMaybePresentDefaultBrowserPromoForSearchThreshold`).
+    }
+}
+
+// MARK: - Default browser promo after search threshold (MOB-4323)
+// DefaultBrowserViewController was defined but never presented. Wire it to the
+// search-count threshold using a dedicated prefs key (not PrefsKeys.IntroSeen, which belongs to onboarding).
+extension BrowserViewController {
+    /// Ecosia prefs key: set to `true` once the search-threshold default-browser promo has been shown.
+    static let ecosiaDefaultBrowserSearchThresholdPromoShownKey = "EcosiaDefaultBrowserSearchThresholdPromoShown"
```
Comment (lucaschifino) [inline firefox-ios/Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift]: Makes sense to use a specific key, but doesn't that mean we need to migrate the value from the previously used key to the new one? Otherwise users that already saw it might see it again.

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 2]
```diff
@@ -1337,6 +1339,9 @@ class BrowserViewController: UIViewController,
                 onStopDownloads(notificationWindowUUID: windowUUID)
             case .SettingsDismissed:
                 onSettingsDismissed()
+            // Ecosia: Re-evaluate default-browser promo whenever the search count changes (MOB-4323).
+            case .searchesCounterChanged:
+                self.ecosiaMaybePresentDefaultBrowserPromoForSearchThreshold()
```
Comment (lucaschifino) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift]: I think we were previously only showing the banner on the NTP (calling it inside `showEmbeddedHomepage` method), with this being here that won't be the case right?

This might be an issue if it blocks the user mid-flow, for example right after making a search and before clicking on a result. Iirc that was why we did it on NTP previously.

**Review by lucaschifino — COMMENTED** [review]
I think the timing might be not exactly what we want, let me know if I'm missing something

**Comment by d4r1091** [comment]
Hey hey 👋 !
Thanks for reviewing it and for the messages above, which helped in terms of a general refresh of what was done.
The new series of commits brings the following along 👇 

- Trigger the eligibility check in `showEmbeddedHomepage` so the promo is only ever offered when the user lands on the NTP
- Store the promo-shown flag in `User.shared`
- Write the flag in `defaultBrowserDidShow` (fired by `viewDidAppear`) so a crash during the transition doesn't silently suppress the promo forever
- One-shot migration from `PrefsKeys.IntroSeen`: on the first launch of the new build

**Review by lucaschifino — APPROVED** [review]
Looks good now! Thanks for the changes 🙌
